R/run_stress_test.R

```diff
@@ -1,55 +1,126 @@
 #'   range compare `shock_year_range_lookup`.
-#' @param term Numeric. A coefficient that determines for which maturity the
+#' @param term_arg Numeric. A coefficient that determines for which maturity the
 #'   expected loss should be calculated in the credit risk section. For accepted
 #'   range compare `term_range_lookup`.
-#' @param company_exclusion Boolean, indicating if companies provided in dataset
+#' @param company_exclusion_arg Boolean, indicating if companies provided in dataset
 #'   excluded_companies.csv shall be excluded.
 #' @return NULL
 #' @export
-run_stress_test <- function(asset_type,
-                            lgd_senior_claims = 0.45,
-                            lgd_subordinated_claims = 0.75,
-                            terminal_value = 0,
-                            risk_free_rate = 0.02,
-                            discount_rate = 0.02,
-                            div_netprofit_prop_coef = 1,
-                            shock_year = 2030,
-                            term = 2,
-                            company_exclusion = TRUE) {
-
-  cat("-- Validating input arguments. \n")
+run_stress_test <- function(asset_type_arg,
+                            lgd_senior_claims_arg = 0.45,
+                            lgd_subordinated_claims_arg = 0.75,
+                            terminal_value_arg = 0,
+                            risk_free_rate_arg = 0.02,
+                            discount_rate_arg = 0.02,
+                            div_netprofit_prop_coef_arg = 1,
+                            shock_year_arg = 2030,
+                            term_arg = 2,
+                            company_exclusion_arg = TRUE) {
+  cat("Running transition risk stress test \n")
+
+  args_list <- mget(names(formals()), sys.frame(sys.nframe()))
+  iter_var <- get_iter_var(args_list)
+  args_tibble <- tibble::as_tibble(args_list)
+
+  st_results_list <- purrr::map(1:nrow(args_tibble), run_stress_test_iteration, args_tibble = args_tibble)
+
+  result_names <- names(st_results_list[[1]])
+  st_results <- result_names %>%
+    purrr::map(function(tib) {
+      purrr::map_dfr(st_results_list, `[[`, tib)
+    }) %>%
+    purrr::set_names(result_names)
+
+  write_stress_test_results(
+    results = st_results$results,
+    expected_loss = st_results$expected_loss,
+    annual_pd_changes = st_results$annual_pd_changes,
+    overall_pd_changes = st_results$overall_pd_changes,
+    asset_type = asset_type_arg,
+    calculation_level = calculation_level_lookup,
+    sensitivity_analysis_vars = names(args_list),
+    iter_var = iter_var
+  )
+
+  cat("-- Exported results to designated output path. \n")
+}
+
+#' Iterate over stress test runs
+#'
+#' @param n Numeric.
+#' @param args_tibble  A tibble holding a set of params for
+#'   `run_stress_test_imp` per row.
+#'
+#' @return List of stress test results.
+run_stress_test_iteration <- function(n, args_tibble) {
```

Comment:
I'll review this in debugger mode to see if I can figure out a way to avoid the `_arg` suffix. There may be something reusable [here](https://github.com/2DegreesInvesting/r2dii.climate.stress.test/pull/166/files#r744294716).